libraries/lib-ipc/IPCChannel.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <cstddef>
+
+/**
+ * \brief Interface for sending data from client to server or vise versa,
+ * complemented by IPCChannelStatusCallback
+ */
+class IPC_API IPCChannel
+{
+public:
+   virtual ~IPCChannel();
+   /**
+    * \brief Write data to the channel
+    * \param bytes Pointer to the data
+    * \param length Number of bytes to send
+    */
+   virtual void Send(const void* bytes, size_t length) = 0;
+};
+
+/**
+ * \brief Interface for listening connection status changes
+ */
+class IPC_API IPCChannelStatusCallback
+{
+public:
+   virtual ~IPCChannelStatusCallback();
+
+   /**
+    * \brief Called when connection attempt fails.
+    */
+   virtual void OnConnectionError() = 0;
+   /**
+    * \brief Called when connection established.
+    * \param channel Using this channel client or server can send data to the other side.
+    */
+   virtual void OnConnect(IPCChannel* channel) = 0;
```

Comment:
Is the argument never null?

If so, I would prefer to use a reference to say what I mean with types, leaving no doubt in the reader's mind about that, while needing no extra comments either.

Reply:
In fact the unique use of OnConnect in BufferedIPCChannel.cpp is always passing non-null (`this`).